I don't think this is the behaviour I'm seeing. E.g. if I run 
```
import nibabel as nib
obj = nib.load('test_data/metab.nii.gz')
print(obj.header.extensions)
print(obj.header.extensions[0])
print(type(obj.header.extensions[0].content))
print(type(obj.header.extensions[0].text))
print(type(obj.header.extensions[0].json()))
obj.header.extensions[0].get_content()
```
I get
```
Nifti1Extensions(NiftiExtension(mrs, b'{"SpectrometerFrequency": [297.219948]...
NiftiExtension(mrs, b'{"SpectrometerFrequency": [297.219948], ...
<class 'bytes'>
<class 'str'>
<class 'dict'>
```
(I've truncate the first two outputs)
but then the final call gives
```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
Cell In[23], [line 8](vscode-notebook-cell:?execution_count=23&line=8)
      [6](vscode-notebook-cell:?execution_count=23&line=6) print(type(obj.header.extensions[0].text))
      [7](vscode-notebook-cell:?execution_count=23&line=7) print(type(obj.header.extensions[0].json()))
----> [8](vscode-notebook-cell:?execution_count=23&line=8) obj.header.extensions[0].get_content()

File ~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:459, in NiftiExtension.get_object(self)
    [452](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:452) """Return the extension content in its runtime representation.
    [453](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:453) 
    [454](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:454) This method may return a different type for each extension type.
    [455](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:455) For simple use cases, consider using ``.content``, ``.text`` or ``.json()``
    [456](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:456) instead.
    [457](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:457) """
    [458](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:458) if self._object is None:
--> [459](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:459)     self._object = self._unmangle(self._raw)
    [460](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:460) return self._object

File ~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:392, in NiftiExtension._unmangle(self, content)
    [391](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:391) def _unmangle(self, content: bytes) -> T:
--> [392](https://file+.vscode-resource.vscode-cdn.net/Users/wclarke/Documents/Python/nifti_mrs_tools/tests/~/miniforge3/envs/nibabel_test_531/lib/python3.12/site-packages/nibabel/nifti1.py:392)     raise NotImplementedError

NotImplementedError:
```

--- a/.github/workflows/push_pr.yml
+++ b/.github/workflows/push_pr.yml
@@ -12,7 +12,7 @@ jobs:
     strategy:
         matrix:
             os: ["ubuntu-latest"]
-            python-version: ["3.8", "3.9", "3.10", "3.11"]
+            python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
 
     steps:
       - uses: actions/checkout@v3

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@ This document contains the nifti_mrs_tools release history in reverse chronologi
 
 1.3.1 (Monday 21st October 2024)
 --------------------------------
+- Drop support for python 3.8. Testing extended to python 3.12 and 3.13.
 - Fix compatibility issue with nibabel > 5.3
 
 1.3.0 (Wednesday 26th June 2024)

--- a/setup.cfg
+++ b/setup.cfg
@@ -9,10 +9,11 @@ license = BSD 3-Clause License
 license_files = LICENSE
 classifiers =
     Programming Language :: Python :: 3
-    Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
     Programming Language :: Python :: 3.11
+    Programming Language :: Python :: 3.12
+    Programming Language :: Python :: 3.13
 author = William Clarke
 author_email = william.clarke@ndcn.ox.ac.uk
 
@@ -22,7 +23,7 @@ install_requires =
     nibabel>=5.3
     fslpy
     importlib_resources;python_version<'3.10'
-python_requires = >=3.8
+python_requires = >=3.9
 packages = find_namespace:
 package_dir =
     = src

--- a/src/nifti_mrs/tools/split_merge.py
+++ b/src/nifti_mrs/tools/split_merge.py
@@ -40,7 +40,7 @@ def split(nmrs, dimension, index_or_indices):
         if dimension > (nmrs.ndim - 1) or dimension < 4:
             raise ValueError('Dimension must be one of 4, 5, or 6 (or DIM_TAG string).'
                              f' This data has {nmrs.ndim} dimensions,'
-                             f' i.e. a maximum dimension value of {nmrs.ndim-1}.')
+                             f' i.e. a maximum dimension value of {nmrs.ndim - 1}.')
         dim_index = dimension
     else:
         raise TypeError('Dimension must be an int (4, 5, or 6) or string (DIM_TAG string).')
@@ -164,7 +164,7 @@ def merge(array_of_nmrs, dimension):
         if dimension > (array_of_nmrs[0].ndim - 1) or dimension < 4:
             raise ValueError('Dimension must be one of 4, 5, or 6 (or DIM_TAG string).'
                              f' This data has {array_of_nmrs[0].ndim} dimensions,'
-                             f' i.e. a maximum dimension value of {array_of_nmrs[0].ndim-1}.')
+                             f' i.e. a maximum dimension value of {array_of_nmrs[0].ndim - 1}.')
         dim_index = dimension
     else:
         raise TypeError('Dimension must be an int (4, 5, or 6) or string (DIM_TAG string).')

